src/lib.rs

```diff
@@ -3100,3 +3161,24 @@ fn map_darwin_target_from_rust_to_compiler_architecture(target: &str) -> Option<
         None
     }
 }
+
+fn which(tool: &PathBuf) -> Option<PathBuf> {
```

Comment:
FWIW when taking an argument by reference, `&Path` is typically preferred over `&PathBuf` (not that it'll have any real impact here, just idioms)